cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -615,8 +616,104 @@ SwiftGenerator::writeServantDocSummary(IceInternal::Output& out, const Interface
 void
 SwiftGenerator::validateMetadata(const UnitPtr& u)
 {
-    MetadataVisitor visitor;
-    u->visit(&visitor);
+    map<string, MetadataInfo> knownMetadata;
+
+    // "swift:attribute"
+    MetadataInfo attributeInfo = {
+        .validOn = {typeid(ClassDecl), typeid(Struct), typeid(Slice::Exception), typeid(Enum)},
+        .acceptedArgumentKind = MetadataArgumentKind::RequiredTextArgument,
+        .mustBeUnique = false,
+    };
+    knownMetadata.emplace("swift:attribute", attributeInfo);
+
+    // "swift:class-resolver-prefix"
+    MetadataInfo classResolverPrefixInfo = {
+        .validOn = {typeid(Unit)},
+        .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
+    };
+    knownMetadata.emplace("swift:class-resolver-prefix", classResolverPrefixInfo);
+
+    // "swift:inherits"
+    MetadataInfo inheritsInfo = {
+        .validOn = {typeid(InterfaceDecl)},
+        .acceptedArgumentKind = MetadataArgumentKind::RequiredTextArgument,
+        .mustBeUnique = false,
+    };
+    knownMetadata.emplace("swift:inherits", inheritsInfo);
+
+    // "swift:module"
+    MetadataInfo moduleInfo = {
+        .validOn = {typeid(Module)},
+        // Even though it's really 'module:prefix' the validator sees this as a single argument since there's no commas.
+        .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
```

Comment:
We should drop the MetadataArgumentKind::MultipleArguments (?) option since we don't use it for any Slice metadata except support warnings.